src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs

```diff
@@ -236,7 +237,27 @@ private IEnumerable<Document> GetDocuments()
                 else if (_ids != null)
                 {
                     if (_ids.Count == 0)
+                    {
                         documents = Enumerable.Empty<Document>();
+                    }
+                    else if (_alreadySeenIdsCount != null)
+                    {
+                        var idsLeft = _ids.Count - _alreadySeenIdsCount.Value;
+                        if (idsLeft == 0)
```

Comment:
Can this be negative?

Reply:
Cannot be negative since I'm counting the ids that I have already seen from the request (and not those that are available in the database).